extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/utils/GeometryUtils.java

```diff
@@ -0,0 +1,114 @@
+package mil.nga.giat.geowave.adapter.vector.utils;
+
+import java.awt.geom.Point2D;
+import java.util.Arrays;
+import java.util.Collections;
+
+import javax.measure.unit.SI;
+import javax.measure.unit.Unit;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.log4j.Logger;
+import org.geotools.ows.bindings.UnitBinding;
+import org.geotools.referencing.GeodeticCalculator;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Envelope;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.Point;
+
+public class GeometryUtils
+{
+
+	private static Logger LOGGER = Logger.getLogger(GeometryUtils.class);
+
+	public static final Pair<Geometry, Double> convertToCRS(
+			Geometry geometry,
+			String distanceUnits,
+			double distance ) {
+		Unit<?> unit;
+		try {
+			unit = (Unit<?>) new UnitBinding().parse(
+					null,
+					distanceUnits);
+		}
+		catch (Exception e) {
+			unit = SI.METER;
+			LOGGER.warn(
+					"Cannot lookup unit of measure " + distanceUnits,
+					e);
+		}
+		final double meterDistance = unit.getConverterTo(
+				SI.METER).convert(
+				distance);
+		final double degrees = distanceToDegrees(
+				geometry,
+				meterDistance);
+		return Pair.of(
+				geometry.buffer(degrees),
+				degrees);
+
+	}
+
+	/**
+	 * Convert meters to decimal degrees based on widest point
+	 */
+	private static double distanceToDegrees(
+			Geometry geometry,
+			double meters ) {
+		final GeometryFactory factory = geometry.getFactory();
+		return (geometry instanceof Point) ? geometry.distance(farthestPoint(
+				(Point) geometry,
+				meters)) : distanceToDegrees(
+				geometry.getEnvelopeInternal(),
+				factory == null ? new GeometryFactory() : factory,
+				meters);
+	}
+
+	private static double distanceToDegrees(
+			Envelope env,
+			GeometryFactory factory,
+			double meters ) {
+		return Collections.max(Arrays.asList(
```

Comment:
I'd suggest we just pick a single point, either the centroid or one with the y value with maximum absolute value (nearest the poles).  I think by using a point with the y value nearest a pole you will get the max degree distance from meters so it should be the most inclusive if that is desired (I could see arguments for either case).